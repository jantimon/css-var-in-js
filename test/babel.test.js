import { expect, describe, it } from "vitest";
import { transformFileSync } from "@babel/core";
import path from "path";
import plugin from "../babel";

describe("production transform", () => {
  it("CSSVariable", () => {
    const { code } = transformFileSync(
      path.join(__dirname, "fixtures/CSSVariable.js"),
      {
        plugins: [[plugin]],
        babelrc: false,
      }
    );
    expect(code).toMatchSnapshot();
  });
  it("createVar", () => {
    const { code } = transformFileSync(
      path.join(__dirname, "fixtures/createVar.js"),
      {
        plugins: [[plugin]],
        babelrc: false,
      }
    );
    expect(code).toMatchSnapshot();
  });
  it("renamed", () => {
    const { code } = transformFileSync(
      path.join(__dirname, "fixtures/renamed.js"),
      {
        plugins: [[plugin]],
        babelrc: false,
      }
    );
    expect(code).toMatchSnapshot();
  });
});

describe("development transform", () => {
  it("CSSVariable", () => {
    const { code } = transformFileSync(
      path.join(__dirname, "fixtures/CSSVariable.js"),
      {
        plugins: [[plugin]],
        babelrc: false,
        envName: "development",
      }
    );
    expect(code).toMatchSnapshot();
  });
  it("createVar", () => {
    const { code } = transformFileSync(
      path.join(__dirname, "fixtures/createVar.js"),
      {
        plugins: [[plugin]],
        babelrc: false,
        envName: "development",
      }
    );
    expect(code).toMatchSnapshot();
  });
  it("renamed", () => {
    const { code } = transformFileSync(
      path.join(__dirname, "fixtures/renamed.js"),
      {
        plugins: [[plugin]],
        babelrc: false,
        envName: "development",
      }
    );
    expect(code).toMatchSnapshot();
  });
});

describe("production transform with displayName", () => {
  it("CSSVariable", () => {
    const { code } = transformFileSync(
      path.join(__dirname, "fixtures/CSSVariable.js"),
      {
        plugins: [[plugin, { displayName: true }]],
        babelrc: false,
      }
    );
    expect(code).toMatchSnapshot();
  });
  it("createVar", () => {
    const { code } = transformFileSync(
      path.join(__dirname, "fixtures/createVar.js"),
      {
        plugins: [[plugin, { displayName: true }]],
        babelrc: false,
      }
    );
    expect(code).toMatchSnapshot();
  });
  it("renamed", () => {
    const { code } = transformFileSync(
      path.join(__dirname, "fixtures/renamed.js"),
      {
        plugins: [[plugin, { displayName: true }]],
        babelrc: false,
      }
    );
    expect(code).toMatchSnapshot();
  });
});
