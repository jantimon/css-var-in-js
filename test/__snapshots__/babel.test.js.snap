// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`development transform > CSSVariable 1`] = `
"import { CSSVariable } from \\"css-variable\\";
const primary = /*@__PURE__*/new CSSVariable(\\"primary--mobtag0\\");
const secondary = /*@__PURE__*/new CSSVariable(\\"secondary--mobtag1\\", {
  value: '#fff'
});
const theme = {
  colors: {
    primary: /*@__PURE__*/new CSSVariable(\\"primary--mobtag2\\")
  }
};
const generated = allColors.map(color => /*@__PURE__*/new CSSVariable(\\"mobtag3\\", color));
const generatedWithFallback = allColors.map(color => /*@__PURE__*/new CSSVariable(\\"mobtag4\\", color, {
  value: '#000'
}));"
`;

exports[`development transform > createVar 1`] = `
"import { createVar } from \\"css-variable\\";
const primary = /*@__PURE__*/createVar(\\"primary--1c6m1ot0\\");
const secondary = /*@__PURE__*/createVar(\\"secondary--1c6m1ot1\\", {
  value: '#fff'
});
const theme = {
  colors: {
    primary: /*@__PURE__*/createVar(\\"primary--1c6m1ot2\\")
  }
};
const generated = allColors.map(color => /*@__PURE__*/createVar(\\"1c6m1ot3\\", color));
const generatedWithFallback = allColors.map(color => /*@__PURE__*/createVar(\\"1c6m1ot4\\", color, {
  value: '#000'
}));"
`;

exports[`development transform > renamed 1`] = `
"import { CSSVariable as CSSVar, createVar as createVariable } from \\"css-variable\\";
const primary = /*@__PURE__*/new CSSVar(\\"primary--1ynmbls0\\");
const secondary = /*@__PURE__*/createVariable(\\"secondary--1ynmbls1\\");"
`;

exports[`production transform > CSSVariable 1`] = `
"import { CSSVariable } from \\"css-variable\\";
const primary = /*@__PURE__*/new CSSVariable(\\"mobtag0\\");
const secondary = /*@__PURE__*/new CSSVariable(\\"mobtag1\\", {
  value: '#fff'
});
const theme = {
  colors: {
    primary: /*@__PURE__*/new CSSVariable(\\"mobtag2\\")
  }
};
const generated = allColors.map(color => /*@__PURE__*/new CSSVariable(\\"mobtag3\\", color));
const generatedWithFallback = allColors.map(color => /*@__PURE__*/new CSSVariable(\\"mobtag4\\", color, {
  value: '#000'
}));"
`;

exports[`production transform > createVar 1`] = `
"import { createVar } from \\"css-variable\\";
const primary = /*@__PURE__*/createVar(\\"1c6m1ot0\\");
const secondary = /*@__PURE__*/createVar(\\"1c6m1ot1\\", {
  value: '#fff'
});
const theme = {
  colors: {
    primary: /*@__PURE__*/createVar(\\"1c6m1ot2\\")
  }
};
const generated = allColors.map(color => /*@__PURE__*/createVar(\\"1c6m1ot3\\", color));
const generatedWithFallback = allColors.map(color => /*@__PURE__*/createVar(\\"1c6m1ot4\\", color, {
  value: '#000'
}));"
`;

exports[`production transform > renamed 1`] = `
"import { CSSVariable as CSSVar, createVar as createVariable } from \\"css-variable\\";
const primary = /*@__PURE__*/new CSSVar(\\"1ynmbls0\\");
const secondary = /*@__PURE__*/createVariable(\\"1ynmbls1\\");"
`;

exports[`production transform with displayName > CSSVariable 1`] = `
"import { CSSVariable } from \\"css-variable\\";
const primary = /*@__PURE__*/new CSSVariable(\\"primary--mobtag0\\");
const secondary = /*@__PURE__*/new CSSVariable(\\"secondary--mobtag1\\", {
  value: '#fff'
});
const theme = {
  colors: {
    primary: /*@__PURE__*/new CSSVariable(\\"primary--mobtag2\\")
  }
};
const generated = allColors.map(color => /*@__PURE__*/new CSSVariable(\\"mobtag3\\", color));
const generatedWithFallback = allColors.map(color => /*@__PURE__*/new CSSVariable(\\"mobtag4\\", color, {
  value: '#000'
}));"
`;

exports[`production transform with displayName > createVar 1`] = `
"import { createVar } from \\"css-variable\\";
const primary = /*@__PURE__*/createVar(\\"primary--1c6m1ot0\\");
const secondary = /*@__PURE__*/createVar(\\"secondary--1c6m1ot1\\", {
  value: '#fff'
});
const theme = {
  colors: {
    primary: /*@__PURE__*/createVar(\\"primary--1c6m1ot2\\")
  }
};
const generated = allColors.map(color => /*@__PURE__*/createVar(\\"1c6m1ot3\\", color));
const generatedWithFallback = allColors.map(color => /*@__PURE__*/createVar(\\"1c6m1ot4\\", color, {
  value: '#000'
}));"
`;

exports[`production transform with displayName > renamed 1`] = `
"import { CSSVariable as CSSVar, createVar as createVariable } from \\"css-variable\\";
const primary = /*@__PURE__*/new CSSVar(\\"primary--1ynmbls0\\");
const secondary = /*@__PURE__*/createVariable(\\"secondary--1ynmbls1\\");"
`;
